spec/models/validations/sales/financial_validations_spec.rb

```diff
@@ -160,6 +160,118 @@
+      expect(record.errors["type"]).to eq(["The minimum increase in equity while staircasing is 10% for this shared ownership type"])
+      record.errors.clear
+
+      record.type = 18
+      financial_validator.validate_percentage_bought_at_least_threshold(record)
+      expect(record.errors["stairbought"]).to eq(["The minimum increase in equity while staircasing is 10%"])
+      expect(record.errors["type"]).to eq(["The minimum increase in equity while staircasing is 10% for this shared ownership type"])
+      record.errors.clear
+
+      record.type = 24
+      financial_validator.validate_percentage_bought_at_least_threshold(record)
+      expect(record.errors["stairbought"]).to eq(["The minimum increase in equity while staircasing is 10%"])
+      expect(record.errors["type"]).to eq(["The minimum increase in equity while staircasing is 10% for this shared ownership type"])
+      record.errors.clear
+
+      record.stairbought = 0
+
+      record.type = 28
+      financial_validator.validate_percentage_bought_at_least_threshold(record)
+      expect(record.errors["stairbought"]).to eq(["The minimum increase in equity while staircasing is 1%"])
+      expect(record.errors["type"]).to eq(["The minimum increase in equity while staircasing is 1% for this shared ownership type"])
+      record.errors.clear
+
+      record.type = 30
+      financial_validator.validate_percentage_bought_at_least_threshold(record)
+      expect(record.errors["stairbought"]).to eq(["The minimum increase in equity while staircasing is 1%"])
+      expect(record.errors["type"]).to eq(["The minimum increase in equity while staircasing is 1% for this shared ownership type"])
+      record.errors.clear
+
+      record.type = 31
+      financial_validator.validate_percentage_bought_at_least_threshold(record)
+      expect(record.errors["stairbought"]).to eq(["The minimum increase in equity while staircasing is 1%"])
+      expect(record.errors["type"]).to eq(["The minimum increase in equity while staircasing is 1% for this shared ownership type"])
+      record.errors.clear
+
+      record.type = 32
+      financial_validator.validate_percentage_bought_at_least_threshold(record)
+      expect(record.errors["stairbought"]).to eq(["The minimum increase in equity while staircasing is 1%"])
+      expect(record.errors["type"]).to eq(["The minimum increase in equity while staircasing is 1% for this shared ownership type"])
+      record.errors.clear
+    end
+
+    it "doesn't add an error to stairbought and type if the percentage bought is less than the threshold (which is 1% by default, but higher for some shared ownership types)" do
+      record.stairbought = 10
+
+      record.type = 2
+      financial_validator.validate_percentage_bought_at_least_threshold(record)
+      expect(record.errors["stairbought"]).to be_empty
+      expect(record.errors["type"]).to be_empty
```

Comment:
for all of these tests (sidenote, I'm not sure you need to test every one, but it doesn't hurt I suppose) you can replace these two lines with just `expect(record.errors).to be_empty`